make maximum coolant available to a crew configurable

czenker opened the pull request:
I think adding different coolant levels might make some missions more interesting:

* @TolotosRhodan created a [mission](https://github.com/TolotosRhodan/EE-MotU) where you can dispose of some coolant in order to create a temporary nebula to escape like a ninja
* decreasing coolant would force a crew to play more strategic and turn systems off more often
* scenarios could add a mechanic where coolant has to be refilled regularly or can be earned as reward
* "sabotage" the players by draining their coolant and have them improvise

![EmptyEpsilon_015](https://user-images.githubusercontent.com/264799/61749434-111e8c80-ada3-11e9-9e0e-7ac914ee4db7.png)
![EmptyEpsilon_016](https://user-images.githubusercontent.com/264799/61749440-14197d00-ada3-11e9-9f37-08964d0c632a.png)

---

Im currently having a problem with linking. The compiler runs fine, but the linker seems not to be able to find the reference to `max_coolant_per_system`. I spend over an hour trying to figure out what the problem is, but I am out of ideas.

`playerSpaceship.cpp:(.text+0x5876): undefined reference to 
 PlayerSpaceship::max_coolant_per_system'`

Maybe someone can give it a try. It could be an issue with my setup or it is just my missing C++ knowledge or some weird foo... :-/ 

@daid to the rescue!

czenker commented:
I found the reason for the linker issue: https://stackoverflow.com/questions/16957458/static-const-in-c-class-undefined-reference/16957554 It had to do with the way `std::min` works.

daid commented:
Yeah, that linking issue is really annoying. It's fixed in the C++17 or C++19 standard (not sure which)

I like the feature. A scenario with a coolant leak sounds like fun, puts some pressure on the players.
Not sure about the name and location of the "total coolant" label. It's more of a global information thing for the engineer, now located at a system specific slider.

Maybe "Available coolant: 100%" would be a better label, and on where to put it, not sure, the screen is already quite full... and I think the label should also be placed on the power/coolant screen as well.

Showing the amount of used coolant doesn't do much I think. As the total used is 0 or 100% of available due to the automatic distribution system.

czenker commented:
Good suggestions. I moved the display to the top left on the engineering screen. This really makes it look less clumsy.

![EmptyEpsilon_018](https://user-images.githubusercontent.com/264799/61893079-85753f00-af0d-11e9-8a02-f1e011d7beaf.png)

Also I added it to the Power Management Screen. I'm not 100% happy with the looks but I think it will do. To make the coolant display less lost I also added the energy display – I think it makes sense to also have it there.

![EmptyEpsilon_020](https://user-images.githubusercontent.com/264799/61893259-f288d480-af0d-11e9-98d0-2d5616ae3795.png)

And finally this is the test script I used for playing around.

    -- Name: Hello World
    -- Description: Testing stuff
    -- Type: Mission
    
    require "utils.lua"
    function init()
        local p = PlayerSpaceship():setTemplate("Flavia P.Falcon"):setFaction("Human Navy")
    
        p:addCustomButton("engineering", "up", "coolant +2", function()
            p:setMaxCoolant(p:getMaxCoolant() + 2)
            print(p:getMaxCoolant())
        end)
        p:addCustomButton("engineering", "down", "coolant -2", function()
            p:setMaxCoolant(p:getMaxCoolant() - 2)
            print(p:getMaxCoolant())
        end)
    
        local auto_coolant = true;
    
        p:addCustomButton("engineering", "coolant", "toggle autocoolant", function()
            p:setAutoCoolant(auto_coolant)
            auto_coolant = not auto_coolant
        end)
    end

daid commented:
Yeah, much better I think. I also like the addition to the power management screen.
